pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -89,8 +92,67 @@ func fileContent(filename string) ([]byte, error) {
 	return content, nil
 }
 
-func askCredentials() (Config, error) {
-	EngineConfig := Config{}
+// askPassword will ask the password to connect to Engine API.
+// The password provided will be added in the Config struct.
+// If an error happens, it will ask again username for users.
+func askPassword() error {
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "oVirt engine password",
+				Help:    "",
+			},
+			Validate: survey.ComposeValidators(survey.Required, authenticated(&EngineConfig)),
+		},
+	}, &EngineConfig.Password)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// askUsername will ask username to connect to Engine API.
+// The username provided will be added in the Config struct.
+// Returns Config and error if failure.
+func askUsername() error {
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "oVirt engine username",
+				Help:    "The user must have permissions to create VMs and disks on the Storage Domain with the same name as the OpenShift cluster.",
+				Default: "admin@internal",
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, &EngineConfig.Username)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// askCredentials will handle username and password for connecting with Engine
+// In case of error during password, users will be prompted username again.
+func askCredentials() error {
+
+	err := askUsername()
```

Comment:
This will give the user one chance at correcting the username but not the password -> which can be annoying if you have entered the wrong password, then you are asked only for the username and fails authentication.

Plus in case someone is trying to hack into the account it will help him find the username in case he has the password --> It is generally good practice security to not let the user know which credentials are wrong.

you can try and add retries (3 for example)

Reply:
> This will give the user one chance at correcting the username but not the password -> which
> can be annoying if you have entered the wrong password, then you are asked only for the
> username and fails authentication.

humm, I don't follow. Have you tried? For me, after this patch is like:

user> foobar
pass> ***********   <---- wrong 
       -> It will keep asking password until you press CTRL-C 
                        (which is the chance to change the username)

> Plus in case someone is trying to hack into the account it will help him find the username in 
> case he has the password --> It is generally good practice security to not let the user know
>  which credentials are wrong.
> you can try and add retries (3 for example)

This is the current behavior as well and I agree must be changed.  However, I won't worry too much about security attempts because Engine has a mechanism of locking username which tries 'X' times to login with incorrect password. Anyway, I think would be nice to avoid the lock, giving a user 2 times or max 3 to tries. :)

Reply:
@Gal-Zaidman  I understood what you meant, the go library survey1 that we use, allow us to keep trying the password until ctrl c is hit. Well, in that case, the user will be blocked by engine if 3 wrong passwords are provided. I think it's something we can research and talk with OCP guys as they use this library as well but I don't think it's a blocker in this PR. I added a new patch related to this topic just in case.